### DISCUSSION TITLE
Android HttpClient timeout Java.Net.SocketException

### DISCUSSION THREAD

**Discussion by angu-hsi** [xamarin/Xamarin.Forms#15793] [Q&A]
Hi Xamarin.Forms Community,

I'm trying to figure out if the following behaviour is common to all and how other developers managed this kind of situation.

In my application I use HttpClient to make request to a web service; the HttpClient object is created with a specific timeout (example 15 seconds).

If the server doesn't respond in the amount of time specified by the Timeout property an exception is thrown.

The problem is that the exception thrown is of type Java.Net.SocketException, and I cannot catch this specific exception. I could catch only with generic type "Exception".

So the question are:

1. It is correct that a Java exception is thrown for timeout expiration?
2. I would like to identify the timeout error, but how can I do it?

Thanks

**Comment by ZakariaBakhkhouch** [comment]
1 - The exception should be System.Net.Http.HttpRequestException, as you use HttpClient to make a request to a web service.
2 - To dentify the timeout error when using HttpClient, you can try to catch the System.Net.Http.HttpRequestException : 
try
{
    // your code
}
catch (HttpRequestException ex)
{
    if (ex.InnerException is System.Net.Sockets.SocketException socketException)
    {
        // Catch the timeout error
        Console.WriteLine("Timeout error:" + ex.Message);
    }
    else
    {
        // Handle other HttpRequestException types here
        Console.WriteLine("Other request exception: " + ex.Message);
    }
}
catch (Exception ex)
{
    // Handle other exceptions that are not HttpRequestException
    Console.WriteLine("General exception: " + ex.Message);
}
